shell-eio.nix

```diff
@@ -0,0 +1,29 @@
+{ pkgs ? import (fetchTarball
+  "https://github.com/NixOS/nixpkgs/archive/11fbbb2126ebb6c83b85908a7773979cbfe270d2.tar.gz")
+  { } }:
+
+pkgs.mkShell {
+  buildInputs = with pkgs; [
+    autoconf
+    rsync
+    git
+    m4
+    patch
+    unzip
+    wget
+    pkg-config
+    gcc
+    gmp
+    libev
+    hidapi
+    libffi
+    jq
+    zlib
+    rustup
+    opam
+    bc
+    httpie
+  ];
```

Comment:
I don't think we need most of these packages

Reply:
Yes, agreed. The whole file `shell-eio.nix` is there because I don't know how to use/install trunk ocaml 5.00 via the nix package system. trunk ocaml 5.00 because it has multicore/effect support. I am a newly minted newbie on nixos so the nix packaging system, nix language are all alien to me now.